solr/core/src/test/org/apache/solr/cli/ZkSubcommandsTest.java

```diff
@@ -406,45 +407,35 @@ public void testUpConfigLinkConfigClearZk() throws Exception {
       assertEquals("Comparing downloaded files to what is in ZK", files.size(), zkFiles.size());
     }
 
-    Path sourceConfDir = ExternalPaths.TECHPRODUCTS_CONFIGSET;
-    // filter out all directories starting with . (e.g. .svn)
-    try (Stream<Path> stream = Files.walk(sourceConfDir)) {
-      List<Path> files =
-          stream
-              .filter(Files::isRegularFile)
-              .filter(path -> path.getFileName().startsWith("."))
-              .toList();
-      files.forEach(
-          (sourceFile) -> {
-            int indexOfRelativePath =
-                sourceFile
-                    .toString()
-                    .lastIndexOf(
-                        "sample_techproducts_configs"
-                            + sourceFile.getFileSystem().getSeparator()
-                            + "conf");
-            String relativePathofFile = sourceFile.toString().substring(indexOfRelativePath + 33);
-            Path downloadedFile = confDir.resolve(relativePathofFile);
-            if (ConfigSetService.UPLOAD_FILENAME_EXCLUDE_PATTERN
-                .matcher(relativePathofFile)
-                .matches()) {
-              assertFalse(
-                  sourceFile + " exists in ZK, downloaded:" + downloadedFile,
-                  Files.exists(downloadedFile));
-            } else {
-              assertTrue(
-                  downloadedFile + " does not exist source:" + sourceFile,
-                  Files.exists(downloadedFile));
-              try {
-                assertEquals(
-                    relativePathofFile + " content changed",
-                    -1,
-                    Files.mismatch(sourceFile, downloadedFile));
-              } catch (IOException e) {
-                throw new RuntimeException(e);
-              }
-            }
-          });
+    // compare the original source with the downloaded destination
+    try (Stream<Path> stream = Files.walk(ExternalPaths.TECHPRODUCTS_CONFIGSET)) {
+      stream
+          .filter(Files::isRegularFile)
+          .forEach(
+              (sourceFile) -> {
+                Path sourceFileRelative =
+                    ExternalPaths.TECHPRODUCTS_CONFIGSET.relativize(sourceFile);
+                Path downloadedFile = destDir.resolve("conf").resolve(sourceFileRelative);
```

Comment:
this was a huge mistake I fixed.  The source and destination were the same!  The destination *should* have been at the temp dir but it was a generically worded "confDir" that just so happened to be `ExternalPaths.TECHPRODUCTS_CONFIGSET` (the source) (!).  Thus the test was very flawed if it even ran, but it didn't as I explained all files were filtered out.